--- realmd-0.14.5/service/realm-credential.h.orig	2013-05-02 14:28:20.000000000 +0200
+++ realmd-0.14.5/service/realm-credential.h	2014-12-21 20:31:06.884885895 +0100
@@ -19,7 +19,7 @@
 
 #include <gio/gio.h>
 
-#include <krb5/krb5.h>
+#include <krb5.h>
 
 G_BEGIN_DECLS
 
--- realmd-0.17.1/service/realm-kerberos.c.orig	2022-09-16 12:57:45.000000000 +0200
+++ realmd-0.17.1/service/realm-kerberos.c	2025-03-02 18:04:10.198432068 +0100
@@ -31,7 +31,7 @@
 #include "realm-provider.h"
 #include "realm-settings.h"
 
-#include <krb5/krb5.h>
+#include <krb5.h>
 
 #include <glib/gi18n.h>
 #include <glib/gstdio.h>
@@ -1082,7 +1082,7 @@ gboolean
 realm_kerberos_flush_keytab (const gchar *realm_name,
                              GError **error)
 {
-	char kt_name[MAX_KEYTAB_NAME_LEN];
+	char kt_name[1100];
 	krb5_error_code code;
 	krb5_keytab keytab;
 	krb5_context ctx;
@@ -1142,12 +1142,11 @@ realm_kerberos_get_netbios_name_from_key
 	krb5_error_code code;
 	krb5_keytab keytab = NULL;
 	krb5_context ctx;
-	krb5_kt_cursor cursor = NULL;
+	krb5_kt_cursor cursor;
 	krb5_keytab_entry entry;
 	krb5_principal realm_princ = NULL;
 	gchar *princ_name = NULL;
 	gchar *netbios_name = NULL;
-	krb5_data *name_data;
 
 	code = krb5_init_context (&ctx);
 	if (code != 0) {
@@ -1169,11 +1168,10 @@ realm_kerberos_get_netbios_name_from_key
 	if (code == 0) {
 		while (!krb5_kt_next_entry (ctx, keytab, &entry, &cursor) && netbios_name == NULL) {
 			if (krb5_realm_compare (ctx, realm_princ, entry.principal)) {
-				name_data = krb5_princ_component (ctx, entry.principal, 0);
+				const char *name_data = krb5_principal_get_comp_string(ctx, entry.principal, 0);
 				if (name_data != NULL
-				                && name_data->length > 0
-				                && name_data->data[name_data->length - 1] == '$') {
-					netbios_name = g_strndup (name_data->data, name_data->length - 1);
+				                && name_data[strlen(name_data) - 1] == '$') {
+					netbios_name = g_strdup (name_data);
 					if (netbios_name == NULL) {
 						code = krb5_kt_free_entry (ctx, &entry);
 						warn_if_krb5_failed (ctx, code);
--- realmd-0.14.5/service/realm-kerberos.h.orig	2013-07-19 17:53:39.000000000 +0200
+++ realmd-0.14.5/service/realm-kerberos.h	2014-12-21 20:31:06.881552562 +0100
@@ -19,7 +19,7 @@
 
 #include <gio/gio.h>
 
-#include <krb5/krb5.h>
+#include <krb5.h>
 
 #include "realm-dbus-generated.h"
 #include "realm-disco.h"
--- realmd-0.14.5/service/realm-kerberos-membership.h.orig	2013-05-02 14:28:20.000000000 +0200
+++ realmd-0.14.5/service/realm-kerberos-membership.h	2014-12-21 20:31:06.884885895 +0100
@@ -19,7 +19,7 @@
 
 #include <gio/gio.h>
 
-#include <krb5/krb5.h>
+#include <krb5.h>
 
 #include "realm-credential.h"
 #include "realm-dbus-generated.h"
--- realmd-0.14.5/tools/realm-client.c.orig	2013-07-24 16:52:32.000000000 +0200
+++ realmd-0.14.5/tools/realm-client.c	2014-12-21 21:54:40.281342170 +0100
@@ -25,7 +25,7 @@
 #include <glib/gstdio.h>
 #include <glib-unix.h>
 
-#include <krb5/krb5.h>
+#include <krb5.h>
 
 #include <sys/socket.h>
 
@@ -664,8 +664,10 @@ kinit_to_ccache (krb5_context krb5,
 	code = krb5_get_init_creds_opt_alloc (krb5, &options);
 	g_return_val_if_fail (code == 0, FALSE);
 
+#ifdef MIT_KRB
 	code = krb5_get_init_creds_opt_set_out_ccache (krb5, options, ccache);
 	g_return_val_if_fail (code == 0, FALSE);
+#endif
 
 	code = krb5_get_init_creds_password (krb5, &my_creds, principal, NULL,
 	                                     krb5_prompter_posix, 0, 0, NULL, options);
@@ -972,7 +974,11 @@ lookup_ccache_credential (const gchar *r
 	                             KRB5_TGS_NAME, realm_name, NULL);
 	g_return_val_if_fail (code == 0, FALSE);
 
+#ifdef MIT_KRB
 	code = krb5_cc_select (krb5, server, &origin, &principal);
+#else
+	code = krb5_cc_cache_match (krb5, &principal, &origin);
+#endif
 
 	krb5_free_principal (krb5, server);
 	if (principal)
--- realmd-0.14.5/tools/realm.c.orig	2013-08-14 17:21:53.000000000 +0200
+++ realmd-0.14.5/tools/realm.c	2014-12-21 20:31:06.834885897 +0100
@@ -23,7 +23,7 @@
 #include <glib/gi18n.h>
 #include <glib-object.h>
 
-#include <krb5/krb5.h>
+#include <krb5.h>
 
 #include <locale.h>
 
--- realmd-0.14.5/tools/realm-discover.c.orig	2013-05-02 14:28:20.000000000 +0200
+++ realmd-0.14.5/tools/realm-discover.c	2014-12-21 20:31:06.861552564 +0100
@@ -19,7 +19,7 @@
 #include "realm-dbus-constants.h"
 #include "realm-dbus-generated.h"
 
-#include <krb5/krb5.h>
+#include <krb5.h>
 
 #include <glib.h>
 #include <glib/gstdio.h>
